core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -101,29 +107,161 @@
   private final boolean inHeap;
 
   /**
+   * The size of memory granted to each consumer.
+   */
+  private final HashMap<MemoryConsumer, Long> consumers;
```

Comment:
This could potentially simplify some of the code in the transferMemory and releaseMemory methods down below.